dev/service-registry/service-locator/src/test/scala/com/lightbend/lagom/registry/impl/ServiceRegistryImplSpec.scala

```diff
@@ -126,6 +126,86 @@ class ServiceRegistryImplSpec extends WordSpecLike with Matchers {
       }
     }
 
+    "can locate 'http' services with or without port-names" in withServiceRegistryActor() { actor =>
+      val registry    = new ServiceRegistryImpl(actor)
+      val serviceName = "fooservice"
+      val expectedUrl = URI.create("http://localhost")
+      registry
+        .register(serviceName)
+        .invoke(
+          ServiceRegistryService.of(
+            expectedUrl,
+            Collections.singletonList(new ServiceAcl(Optional.of(Method.GET), Optional.of("/")))
+          )
+        )
+        .toCompletableFuture
+        .get(testTimeoutInSeconds, TimeUnit.SECONDS)
+
+      withClue("lookup without a portName") {
+        val registeredUrl = registry
+          .lookup(serviceName, Optional.empty())
+          .invoke(NotUsed)
+          .toCompletableFuture
+          .get(
+            testTimeoutInSeconds,
+            TimeUnit.SECONDS
+          )
+        assertResult(expectedUrl)(registeredUrl)
+      }
+
+      withClue("lookup with portName 'http'") {
+        val registeredUrl = registry
+          .lookup(serviceName, Optional.of("http"))
+          .invoke(NotUsed)
+          .toCompletableFuture
+          .get(
+            testTimeoutInSeconds,
+            TimeUnit.SECONDS
+          )
+        assertResult(expectedUrl)(registeredUrl)
+      }
+    }
+
+    "can locate 'https' services only when passing port-name" in withServiceRegistryActor() { actor =>
+      val registry    = new ServiceRegistryImpl(actor)
+      val serviceName = "fooservice"
+      val expectedUrl = URI.create("https://localhost")
+      registry
+        .register(serviceName)
+        .invoke(
+          ServiceRegistryService.of(
+            expectedUrl,
+            Collections.singletonList(new ServiceAcl(Optional.of(Method.GET), Optional.of("/")))
+          )
+        )
+        .toCompletableFuture
+        .get(testTimeoutInSeconds, TimeUnit.SECONDS)
+
+      withClue("lookup without a portName") {
+        val ee = the[ExecutionException] thrownBy registry
+          .lookup(serviceName, Optional.empty())
+          .invoke(NotUsed)
+          .toCompletableFuture
+          .get(
+            testTimeoutInSeconds,
+            TimeUnit.SECONDS
+          )
+        ee.getCause shouldBe a[NotFound]
```

Comment:
This is the consequence of what I wrote above. Once you use `https`, it becomes invisible to the ServiceLocator. In other words, lookup of https for managed service will simply not work.